src/webui/api/torrentscontroller.cpp

```diff
@@ -196,6 +196,33 @@ namespace
 
         return QVariantList {dht, pex, lsd};
     }
+
+    BitTorrent::PeerAddress parsePeer(const QString &peer)
```

Comment:
I tried moving the `PeerAddress` class to `Utils::Net` but I'm having a lot of trouble updating everywhere its used, adding forward declares, etc.

Reply:
@Chocobo1 @glassez I propose we accept the duplicate code for now and refactor in a separate PR. Otherwise someone else will have to jump in to this PR, and I don't see that happening.

Reply:
@Chocobo1, can we accept this, since @Piccirello has problems with the change of C++ code? At least as an exception in this particular case?

Reply:
A TODO comment should be added at relevant places so it won't be forgotten.

Reply:
Okay, I've got some free time to do some routine, so I extracted `PeerAddress` and related `parse()` function into separate file. I still decided to leave it in `BitTorrent` namespace.
@Piccirello, you can easily finish this PR on top of #10963 .